pytorch_lightning/core/lightning.py

```diff
@@ -1891,7 +1891,10 @@ def to_onnx(self, file_path: Union[str, Path], input_sample: Optional[Any] = Non
 
         if "example_outputs" not in kwargs:
             self.eval()
-            kwargs["example_outputs"] = self(input_sample)
+            if isinstance(input_sample, Tuple):
+                kwargs["example_outputs"] = self(*input_sample)
+            else:
+                kwargs["example_outputs"] = self(input_sample)
```

Comment:
@awaelchli are there other places where we handle the example input? I know this is in the model summary
- https://github.com/PyTorchLightning/pytorch-lightning/blob/24f0124ddd478e18b76d64a79f1d7b341a0f8646/pytorch_lightning/utilities/model_summary.py#L291-L299

if so, would it make sense to write a small utility function to accept the lightning module & example inputs and do this checking in one place?

Reply:
no other places in lightning except torchscript, onnx and model summary. 
yes an utility function can be useful even for callbacks etc. outside Lightning which want to use the example inputs.

Reply:
maybe @xerus is interested in following up with a PR for this?